module/gui/Options.py

```diff
@@ -0,0 +1,1500 @@
+        grid.addWidget(self.sbToPort,   2, 1)
+        grid.addWidget(self.cbGetPort,  3, 0, 1, 3)
+        grid.setColumnStretch(2, 1)
+        grid.setRowMinimumHeight(4, 20)
+        grid.addLayout(self.hboxStatus, 5, 0, 1, 3)
+        grid.setRowStretch(6, 1)
+
+        gb = QGroupBox(_("ClickNLoad Port Forwarding") + "     ")
+        gb.setLayout(grid)
+
+        self.buttons = WtDialogButtonBox(Qt.Horizontal, self)
+        self.okBtn     = self.buttons.addButton(QDialogButtonBox.Ok)
+        self.cancelBtn = self.buttons.addButton(QDialogButtonBox.Cancel)
+        self.buttons.button(QDialogButtonBox.Ok).setText(_("OK"))
+        self.buttons.button(QDialogButtonBox.Cancel).setText(_("Cancel"))
+
+        vbox = QVBoxLayout()
+        vbox.addWidget(gb)
+        vbox.addLayout(self.buttons.layout())
+        self.setLayout(vbox)
+
+        self.setMinimumWidth(250)
+        self.adjustSize()
+
+        self.connect(self.okBtn,     SIGNAL("clicked()"), self.accept)
+        self.connect(self.cancelBtn, SIGNAL("clicked()"), self.reject)
+        self.connect(self.cbGetPort, SIGNAL("toggled(bool)"), self.sbToPort.setDisabled)
+
+        # default settings
+        self.settings["enabled"]  = False
+        self.settings["fromIP"]   = "127.0.0.1"
+        self.settings["toIP"]     = "999.999.999.999"
+        self.settings["toPort"]   = 9666
+        self.settings["getPort"]  = False
+        self.defaultFromPort()
+
+    def exec_(self):
+        # It does not resize very well when the font size has changed
+        if self.font() != self.lastFont:
+            self.lastFont = self.font()
+            self.adjustSize()
+        return QDialog.exec_(self)
+
+    def defaultFromPort(self):
+        self.settings["fromPort"] = 9666
+        self.dict2dialogState(True)
+
+    def dialogState2dict(self):
+        self.settings["enabled"]  = self.cbEnable.isChecked()
+        self.settings["fromPort"] = self.sbFromPort.value()
+        self.settings["toPort"]   = self.sbToPort.value()
+        self.settings["getPort"]  = self.cbGetPort.isChecked()
+
+    def dict2dialogState(self, error):
+        self.cbEnable.setChecked(self.settings["enabled"])
+        if (self.settings["fromPort"] < self.sbFromPort.minimum()) or (self.settings["fromPort"] > self.sbFromPort.maximum()):
+            raise ValueError("fromPort out of range") # this is catched by main.loadOptionsFromConfig()
+        self.sbFromPort.setValue(self.settings["fromPort"])
+        #if (self.settings["toPort"] < self.sbToPort.minimum()) or (self.settings["toPort"] > self.sbToPort.maximum()):
+        #    raise ValueError("toPort out of range")
+        self.sbToPort.setValue(self.settings["toPort"])
+        self.cbGetPort.setChecked(self.settings["getPort"])
+        if not error:
+            if self.settings["enabled"]:
+                self.lblStatus.setText(_("Running"))
+            else:
+                self.lblStatus.setText(_("Not Running"))
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['_' may be undefined, or defined from star imports: PyQt4.QtCore, PyQt4.QtGui](https://app.codacy.com/manual/pyLoad/pyload/pullRequest?prid=567903)